Tests for CreateService and annotations

cdrage opened the pull request:
Adds some tests for CreateService, specifically the initial generation
as well as providing an edge case when specifying "kompose.service.type"

cdrage commented:
Fixes 1/2 of https://github.com/kubernetes-incubator/kompose/issues/242

cdrage commented:
Still newbish to Go, so let me know what things I can improve on to make this more idiomatic!

kadel commented on pkg/transformer/kubernetes/k8sutils_test.go:
```diff
@@ -0,0 +1,107 @@
+	objects := k.Transform(kompose_object, kobject.ConvertOptions{CreateD: true, Replicas: 3})
+
+	// Test the creation of the service
+	CreateService("foo", service, objects)
```
Shouldn't be there at least some kind of test that service has been created?
Maybe I just don't understand it, but it looks like this is not testing anything :-)

cdrage replied:
@kadel you're right, i forgot to add the checking err in this test, was initially a copy-and-paste from the test below. thank you!

cdrage commented:
@kadel me neither for the tests, I updated it to check that information is generated. I tried doing a comparison between port numbers (passing 123 and making sure we get 123 back). But I'm having a bit of difficulty digging through k8s code to find the correct way of doing it. Hopefully this update to the PR is sufficient!

kadel commented:
there already is a function for comparing ports in [kubernetes_test.go](https://github.com/kubernetes-incubator/kompose/blob/master/pkg/transformer/kubernetes/kubernetes_test.go#L93)

kadel commented:
you commited vim swap file ;-) -  `pkg/transformer/kubernetes/.k8sutils_test.go.swp`

cdrage commented:
@kadel boo, that's why I added the .gitignore PR earlier xD okay, so i've updated this PR again with new code, although I'm not sure why the tests are erroring.

kadel commented on pkg/transformer/kubernetes/k8sutils_test.go:
```diff
@@ -0,0 +1,110 @@
+
+	// Test the creation of the service with modified annotations (kompose.service.type)
+	svc := k.CreateService("foo", service, objects)
+	if svc.Spec.Type != api.ServiceType("NodePort") {
```
One last thing ;-)
You test only NodePort, it would be great to test also other options for service.type.
This might be great case for something like this: https://github.com/golang/go/wiki/TableDrivenTests

cdrage replied:
@kadel sure, i'll work on that. i'll include it in the top test however since this test is specific to supplying nodeport (testing the annotations / labels :+1: )

cdrage replied:
@kadel so I don't think I can compare anything else (`svc.Spec.Type` is blank, same as `svc.Spec.ClusterIP`, `svc.Spec.LoadBalancerIP`, etc.)

Should I just compare that it comes up as blank?

kadel replied:
@cdrage sorry my comment might be little bit confusing. I should do it as line comment.

What I meant was that it might be nice to make this test more generic. Using table driven tests you can do test that will test all possible values for `kompose.service.type` in one test. Basicaly create more services with different `kompose.service.type` label

cdrage commented:
@kadel updated with your code :) thanks man!

kadel commented:
> @kadel updated with your code :) thanks man!

thanks ;-)
Sorry for confusion, sometimes it is easier for me to express myself in code than in english :laughing:

cdrage commented:
@kadel 

01101110 01101111 00100000 01110000 01110010 01101111 01100010 01101100 01100101 01101101

:)

kadel commented:
00111010 01000100
